--- levmar-2.6/CMakeLists.txt	2011-11-29 13:36:19.000000000 -0800
+++ CMakeLists.txtNEW	2012-02-23 13:12:43.451307537 -0800
@@ -67,7 +67,7 @@
 
 # demo program
 IF(BUILD_DEMO)
-  SET(LIBS levmar)
+  SET(LIBS levmar m)
 
   LINK_DIRECTORIES(${CMAKE_BINARY_DIR}) # location of the levmar library
   LINK_DIRECTORIES(${LAPACKBLAS_DIR})
@@ -94,4 +94,10 @@
   ADD_DEPENDENCIES(lmdemo levmar)
 ENDIF(BUILD_DEMO)
 
+#Setup library installation directories
+file(GLOB HEADERS ${LEVMAR_SOURCE_DIR}/*.h)
+install(FILES ${HEADERS} DESTINATION include/levmar)
+
+install(TARGETS "levmar" ARCHIVE DESTINATION lib)
+
 #SUBDIRS(matlab)
